### PR TITLE
fix: add popular and trending products lists

### PR DESCRIPTION
**Pull request by shahbaz-arbisoft**
### Description

Add popular and trending products to recommendations.

> Related [PR](https://github.com/edx/edx-internal/pull/8902)

#### JIRA

[VAN-1565](https://2u-internal.atlassian.net/browse/VAN-1565)

#### How Has This Been Tested?

Locally

#### Screenshots/sandbox (optional):

Screenshot 2023-07-26 at 5 34 42 PM
Screenshot 2023-07-26 at 5 36 29 PM

### CODE REVIEW

File: src/recommendations/RecommendationsPage.jsx [hunk 1]
```diff
@@ -21,10 +21,29 @@ const RecommendationsPage = (props) => {
 
   const { formatMessage } = useIntl();
   const [selectedRecommendationsType, setSelectedRecommendationsType] = useState(RECOMMENDATIONS_OPTION_LIST[1]);
-  const [recommendations] = useState([]);
+  const [recommendations, setRecommendations] = useState([]);
+  const [popularRecommendations, setPopularRecommendations] = useState([]);
+  const [trendingRecommendations, setTrendingRecommendations] = useState([]);
+
+  useEffect(() => {
```
Comment (mubbsharanwar) [inline src/recommendations/RecommendationsPage.jsx:28]: We don't need this `useEffect`, `popularRecommendations` and `trendingRecommendations` state. Just use the `recommendations` state in the second `useEffect` as you are already doing and inside it should be like this:

```
 const popularProducts = JSON.parse(getConfig().POPULAR_PRODUCTS);
 const trendingProducts = JSON.parse(getConfig().TRENDING_PRODUCTS);
 setRecommendations(option.value === 'trending' ? trendingRecommendations : popularRecommendations);
```

Comment (shahbaz-arbisoft) [reply src/recommendations/RecommendationsPage.jsx]: Addressed.

File: src/config/index.js [hunk 2]
```diff
@@ -29,6 +29,8 @@ const configuration = {
   INFO_EMAIL: process.env.INFO_EMAIL || '',
   ZENDESK_KEY: process.env.ZENDESK_KEY,
   ZENDESK_LOGO_URL: process.env.ZENDESK_LOGO_URL,
+  POPULAR_PRODUCTS: process.env.POPULAR_PRODUCTS,
+  TRENDING_PRODUCTS: process.env.TRENDING_PRODUCTS,
```
Comment (zainab-amir) [inline src/config/index.js:33]: Default value is missing. Also move the along with General recommendations to a separate section above `Miscellaneous`.
```suggestion
  // Recommendation constants
  GENERAL_RECOMMENDATIONS: process.env.GENERAL_RECOMMENDATIONS || '[]',
  POPULAR_PRODUCTS: process.env.POPULAR_PRODUCTS || '[]', 
  TRENDING_PRODUCTS: process.env.TRENDING_PRODUCTS || '[]',
```

Comment (shahbaz-arbisoft) [reply src/config/index.js:33]: Got it. Although, this case is handled within component, I will remove it from there.

Comment (shahbaz-arbisoft) [reply src/config/index.js]: Addressed

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**